src/cube_neon.h

```diff
@@ -1,6 +1,423 @@
+// static cube
+#define static_cube(c_ufr, c_ubl, c_dfl, c_dbr, c_ufl, c_ubr, c_dfr, c_dbl,							\
+					e_uf, e_ub, e_db, e_df, e_ur, e_ul, e_dl, e_dr, e_fr, e_fl, e_bl, e_br)			\
+	((cube_t){																						\
+		.corner = {c_ufr, c_ubl, c_dfl, c_dbr, c_ufl, c_ubr, c_dfr, c_dbl, 0, 0, 0, 0, 0, 0, 0, 0},	\
+		.edge = {e_uf, e_ub, e_db, e_df, e_ur, e_ul, e_dl, e_dr, e_fr, e_fl, e_bl, e_br, 0, 0, 0, 0}})
+
+// zero cube
+#define zero						\
+	(cube_t)						\
+	{								\
+		.corner = vdupq_n_u8(0),	\
+		.edge = vdupq_n_u8(0)		\
+	}
+
+// solved cube
+#define solved static_cube(	\
+	0, 1, 2, 3, 4, 5, 6, 7, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11)
+
+// Functions
+_static void pieces(cube_t *, uint8_t[static 8], uint8_t[static 12]);
+_static_inline bool equal(cube_t, cube_t);
+_static_inline cube_t invertco(cube_t);
+_static_inline cube_t compose_edges(cube_t, cube_t);					// implementation similar to portable compose_edges_inplace
+_static_inline cube_t compose_corners(cube_t, cube_t);					// implementation similar to portable compose_corners_inplace
+_static_inline uint8x16_t compose_edges_slim(uint8x16_t, uint8x16_t);	// similar to compose_edges but without the cube_t struct
+_static_inline uint8x16_t compose_corners_slim(uint8x16_t, uint8x16_t); // similar to compose_corners but without the cube_t struct
+_static_inline cube_t compose(cube_t, cube_t);
+_static_inline cube_t inverse(cube_t);
+
+_static_inline int64_t coord_co(cube_t);
+_static_inline int64_t coord_csep(cube_t);
+_static_inline int64_t coord_cocsep(cube_t);
+_static_inline int64_t coord_eo(cube_t);
+_static_inline int64_t coord_esep(cube_t);
+
+_static_inline void copy_corners(cube_t *, cube_t);
+_static_inline void copy_edges(cube_t *, cube_t);
+_static_inline void set_eo(cube_t *, int64_t);
+_static_inline cube_t invcoord_esep(int64_t);
+
+_static void
+pieces(cube_t *cube, uint8_t c[static 8], uint8_t e[static 12])
+{
+	// First 8 bytes of the corner vector are copied from the c array
+	vst1_u8(c, vget_low_u8(cube->corner));
+
+	// 12 bytes of the edge vector are copied from the e array
+	// First 8 bytes
+	vst1_u8(e, vget_low_u8(cube->edge));
+	// Next 4 bytes
+	vst1_lane_u32((uint32_t *)(e + 8), vreinterpret_u32_u8(vget_high_u8(cube->edge)), 0);
+}
+
+_static_inline bool
+equal(cube_t c1, cube_t c2)
+{
+	uint8x16_t cmp_corner, cmp_edge;
+	uint64x2_t cmp_corner_u64, cmp_edge_u64;
+	uint64x2_t cmp_result;
+
+	// compare the corner vectors
+	cmp_corner = vceqq_u8(c1.corner, c2.corner);
+	// compare the edge vectors
+	cmp_edge = vceqq_u8(c1.edge, c2.edge);
+
+	// convert the comparison vectors to 64-bit vectors
+	cmp_corner_u64 = vreinterpretq_u64_u8(cmp_corner);
+	cmp_edge_u64 = vreinterpretq_u64_u8(cmp_edge);
+
+	// combine the comparison vectors
+	cmp_result = vandq_u64(cmp_corner_u64, cmp_edge_u64);
+
+	// check if all the bits are set
+	return vgetq_lane_u64(cmp_result, 0) == ~0ULL && vgetq_lane_u64(cmp_result, 1) == ~0ULL;
+}
+
+_static_inline cube_t invertco(cube_t c)
```

Comment:
Io di solito metto typo e eventuale `static` su una riga, e nome della funzione e parametri su un'altra riga, come hai fatto per altre funzioni. Falle tutte uguali.

Reply:
Dai prossimi commit farò in modo di fare più attenzione a seguire lo standard.